src/fsharp/FSharp.Build/FSharpEmbedResXSource.fs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.FSharp.Build
+
+open System
+open System.Collections
+open System.Globalization
+open System.IO
+open System.Linq
+open System.Text
+open System.Xml.Linq
+open Microsoft.Build.Framework
+open Microsoft.Build.Utilities
+
+type FSharpEmbedResXSource() =
+    let mutable _buildEngine : IBuildEngine = null
+    let mutable _hostObject : ITaskHost = null
+    let mutable _embeddedText : ITaskItem[] = [||]
+    let mutable _generatedSource : ITaskItem[] = [||]
+    let mutable _outputPath : string = ""
+    let mutable _targetFramework : string = ""
+
+    let boilerplate = @"// <auto-generated>
+
+namespace {0}
+
+open System.Reflection
+
+module internal {1} =
+    type private C (_dummy:System.Object) = class end
+    let mutable Culture = System.Globalization.CultureInfo.CurrentUICulture
+    let ResourceManager = new System.Resources.ResourceManager(""{2}"", C(null).GetType().GetTypeInfo().Assembly)
```

Comment:
Can't this be replaced with `Assembly.GetCallingAssembly`?

Reply:
In which case can we make the `C` class simpler:

```f#
type private Dummy = interface end
```

Reply:
@saul sure.

Reply:
We can't use `typeof<>` because this generator is used by `FSharp.Core` and at the point that we need these strings is before the `typeof<>` operator can be defined  (i.e., in `prim-types.fs/fsi`).  I could do a custom switch on `FSharp.Core`, but that seemed like a nasty hack given that the hack already there will always work.

Reply:
@brettfo thanks for clearing that up, I couldn't for the life of me remember why we did it this way.